Reference/Common-Pitfalls/index.md

```diff
@@ -0,0 +1,466 @@
+with a level of one. As mentioned above, traversing costs resources and in this example there is 3x traversals being done
+for the same value. Instead this can be rewritten as:
+
+```
+@{
+    var site = @Model.Content.Site();
+}
+<ul>
+    <li><a href="@site.Url">@site.Name</a></li>
+    @foreach(var node in site.Children) 
+    {
+        <li><a href="@node.Url">@node.Name</a></li>
+    }
+</ul>
+```
+
+## Dynamics
+
+In Umbraco version 8+ dynamic support for access to IPublishedContent will be removed. 
+There are a few reasons for this:
+
+* Dynamics are much slower than their strongly typed equivalent
+* The codebase for Dynamics is difficult to maintain and it's massive
+* Many querying concepts in Dynamics are difficult to understand and need to be memorize due to all of the string syntax required
+* It is much harder to debug and to know if there are errors since the syntax is not typed or compiled
+* No intellisense is possible inside Visual Studio
+* ModelsBuilder is part of the Umbraco Core and provides much nicer and strongly typed access to property accessors and querying in your views
+
+How do you know if you are using Dynamics?
+
+* If you are using `@CurrentPage` then __you are__ using dynamics
+* If you are using the UmbracoHelper query methods like `@Umbraco.Content` or `@Umbraco.Media` instead of the typed methods like `@Umbraco.TypedContent` and `@Umbraco.TypedMedia` then __you are__ using dynamics
+
+It is strongly advised that you use the strongly typed `@Model.Content` instead of `@CurrentPage` models in your views,  
+this will actually perform much better and you'll be forward compatible with Umbraco v8+ with regards to querying `IPublishedContent`. 
+
+A large problem with the performance of dynamics is having to parse string syntax such as:
+`@CurrentPage.Children.Where("DocumentTypeAlias == \"DatatypesFolder\" && Visible")` and turn that into something that is compilable when 
+instead it can just be written as something that compiles
+
+## Using the Services layer in your views
+
+The Services layer of Umbraco is for manipulating the business logic of Umbraco directly to/from the database. 
+None of these methods should be used within your views and can have a very large impact on performance and stability of 
+your application.
+
+Your views should rely only on the readonly data access of the `UmbracoHelper` and the properties/methods that it exposes. This ensures
+that the data being queried is fast (comes from cache) and that you aren't inadvertently making database changes.
+
+If you are using `Application.Services...` in your views, you should figure out why this is being done and in most cases remove this logic.   
+
+## Using UmbracoContext to access ApplicationContext
+
+You should not access the `ApplicationContext` via the `UmbracoContext`. 
+
+For example: `UmbracoContext.Current.Application` _<-- this is now deprecated/obsolete_
```

Comment:
Should give/explain the reasoning (even thoiugh you kinda cover it the opening of this document) as opposed to just saying, no you shouldn't use this thing...